www/js/controlHelper.ts

```diff
@@ -0,0 +1,128 @@
+import { DateTime } from "luxon";
+
+import { getRawEntries } from "./commHelper";
+import { logInfo, displayError } from "./plugin/logger";
+import i18next from "./i18nextInit" ;
+
+interface fsWindow extends Window {
+  requestFileSystem: (
+    type: number,
+    size: number,
+    successCallback: (fs: any) => void,
+    errorCallback?: (error: any) => void
+  ) => void;
+  LocalFileSystem: {
+    TEMPORARY: number;
+    PERSISTENT: number;
+  };
+};
+
+declare let window: fsWindow;
+
+export const getMyData = function(startTs: Date) {
+    // We are only retrieving data for a single day to avoid
+    // running out of memory on the phone
+    const startTime = DateTime.fromJSDate(startTs);
+    const endTime = startTime.endOf("day");
+    const startTimeString = startTime.toFormat("yyyy'-'MM'-'dd");
+    const endTimeString = endTime.toFormat("yyyy'-'MM'-'dd");
+
+    const dumpFile = startTimeString + "."
+      + endTimeString
+      + ".timeline";
+      alert(`Going to retrieve data to ${dumpFile}`);
+
+    const writeDumpFile = function(result) {
+      const resultList = result.phone_data;
+      return new Promise<void>(function(resolve, reject) {
+        window.requestFileSystem(window.LocalFileSystem.TEMPORARY, 0, function(fs) {
+          console.log(`file system open: ${fs.name}`);
+          fs.root.getFile(dumpFile, { create: true, exclusive: false }, function (fileEntry) {
+            console.log(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`)
+            fileEntry.createWriter(function (fileWriter) {
+              fileWriter.onwriteend = function() {
+                console.log("Successful file write...");
+                resolve();
+              }
+              fileWriter.onerror = function(e) {
+                console.log(`Failed file write: ${e.toString()}`);
+                reject();
+              }
+
+              // if data object is not passed in, create a new blog instead.
+              const dataObj = new Blob([JSON.stringify(resultList, null, 2)],
+                { type: "application/json" });
+              fileWriter.write(dataObj);
+            })
+
+          });
+        });
+      });
+    }
+
+    const emailData = function() {
```

Comment:
I didn't think of that! Yeah, since we expanded the functionality,`shareData` would make more sense at this point.

I had the same apprehension towards writing tests as well - it seemed like a tremendous amount of work to mock and verify email inputs, for mixed returns - but, like you said, I think splitting the function is a great way to tackle it! I'll go ahead and start that now : )